module/core/derive_tools_meta/src/lib.rs

```diff
@@ -520,6 +522,54 @@ pub fn derive_variadic_from( input : proc_macro::TokenStream ) -> proc_macro::To
   }
 }
 
+/// Provides an automatic [Not](core::ops::Not) trait  implementation for struct.
+///
+/// This macro simplifies the creation of a logical negation or complement operation
+/// for structs that encapsulate values which support the `!` operator.
+///
+/// ## Example Usage
+///
+/// Instead of manually implementing [Not](core::ops::Not) for [IsActive]:
+///
+/// ```rust
+/// use core::ops::Not;
+///
+/// pub struct IsActive( bool );
+///
+/// impl Not for IsActive
+/// {
+///   type Output = IsActive;
+///
+///   fn not(self) -> Self::Output
+///   {
+///     IsActive(!self.0)
+///   }
+/// }
+/// ```
+///
+/// Use `#[ derive( Not ) ]` to automatically generate the implementation:
+///
+/// ```rust
+/// # use derive_tools_meta::*;
+/// #[ derive( Not ) ]
+/// pub struct IsActive( bool );
+/// ```
+///
+/// The macro automatically implements the [not](core::ops::Not::not) method, reducing boilerplate code and potential errors.
+///
+#[ cfg( feature = "enabled" ) ]
+#[ cfg( feature = "derive_not" ) ]
+#[ proc_macro_derive( Not, attributes( debug ) ) ]
```

Comment:
should also handle #[ not( off ) ] and #[ not( on ) ]

Reply:
look core\derive_tools\tests\inc\from\variants_duplicates_some_off_default_off.rs